File: Stripe/STPNumericStringValidator.m
//
//  STPNumericStringValidator.m
//  StripeiOS
//
//  Created by Cameron Sabol on 3/6/20.
//  Copyright © 2020 Stripe, Inc. All rights reserved.
//

#import "STPNumericStringValidator.h"

#import "NSCharacterSet+Stripe.h"
#import "NSString+Stripe.h"

NS_ASSUME_NONNULL_BEGIN

@implementation STPNumericStringValidator

+ (BOOL)isStringNumeric:(NSString *)string {
     return [string rangeOfCharacterFromSet:[NSCharacterSet stp_invertedAsciiDigitCharacterSet]].location == NSNotFound;
}

+ (NSString *)sanitizedNumericStringForString:(NSString *)string {
    return [string stp_stringByRemovingCharactersFromSet:[NSCharacterSet stp_invertedAsciiDigitCharacterSet]];
}

@end

NS_ASSUME_NONNULL_END

File: Stripe/STPNumericStringValidator.swift
//
//  STPNumericStringValidator.swift
//  StripeiOS
//
//  Created by Cameron Sabol on 3/6/20.
//  Copyright © 2020 Stripe, Inc. All rights reserved.
//

import Foundation

enum STPTextValidationState: Int {
    case empty
    case incomplete
    case complete
    case invalid
}

class STPNumericStringValidator: NSObject {
    /// Whether or not the target string contains only numeric characters.
    class func isStringNumeric(_ string: String) -> Bool {
        return
            (string as NSString).rangeOfCharacter(from: CharacterSet.stp_invertedAsciiDigit)
            .location
            == NSNotFound
    }

    /// Returns a copy of the passed string with all non-numeric characters removed.
    class func sanitizedNumericString(for string: String) -> String {
        return string.stp_stringByRemovingCharacters(from: CharacterSet.stp_invertedAsciiDigit)
    }
}
